crates/ruff_linter/src/rules/ruff/rules/sequence_sorting.rs

```diff
@@ -0,0 +1,818 @@
+/// Utilities for sorting constant lists of string literals.
+///
+/// Examples where these are useful:
+/// - Sorting `__all__` in the global scope,
+/// - Sorting `__slots__` or `__match_args__` in a class scope
+use std::borrow::Cow;
```

Comment:
Nit: It would have been helpful to extract `sequence_sorting` as part of its own PR or at least commit because there's now no tooling telling me which parts of it are new (and need review) and which parts are unchanged.

Reply:
Sorry about that :(

The reason it ended up like this was that I moved material from `sort_dunder_all.rs` piecemeal -- I wasn't sure at first how much I'd need to move over, so I did it in a slightly ad-hoc way. I appreciate that this made it a messy thing to review, though -- I'll try to do better in the future.